_posts/2018-06-15-cocoapods-keys-react-native.markdown

```diff
@@ -0,0 +1,156 @@
+
+Links to examples below come from [this commit](https://github.com/artsy/emission/pull/1086/commits/4a2a3e9260e97d791536cf38376a06b0ad0946a8) which adds a key for the Stripe API to Emission. We'll be adding this key to an `Emission` module that can be imported to our React Native files.
+
+## Steps
+
+#### 1. Add the key to the app's Podfile.
+
+This is the extent of `cocoapods-keys` official [setup][ck-setup], and after this you **could** set the key via `pod keys set <NAME>` or `pod install`... but we have more to do.
+
+[/Example/Podfile](https://github.com/artsy/emission/blob/4a2a3e9260e97d791536cf38376a06b0ad0946a8/Example/Podfile#L63):
+
+```diff
+plugin 'cocoapods-keys', {
+  :target => 'Emission',
+  :keys => [
+    'ArtsyAPIClientSecret',   # Authing to the Artsy API
+    'ArtsyAPIClientKey',      #
++    'StripePublishableKey',
+  ]
+}
+```
+
+---
+
+#### 2. Configure the library to consume our new key
+
+We'll need to update the `initWithUserId...` function -- one **fun** part of adjusting to Objective-C is that rather than named funtions, we just refer to them by their entire signatures -- to expose the new key as a property and add it to `constantsToExport`.
+
+Note that this is happening in our _Emission Pod_; The pod now expects that key to be available in our _consuming_ Example app as defined above.
+
+[/Pod/Classes/Core/AREmission.h](https://github.com/artsy/emission/blob/4a2a3e9260e97d791536cf38376a06b0ad0946a8/Pod/Classes/Core/AREmission.h#L17-L34):
+
+```diff
+// ENV Variables
++ @property (nonatomic, copy, readonly, nullable) NSString *stripePublishableKey;
+
+# ...
+
+ - (instancetype)initWithUserID:(NSString *)userID
+           authenticationToken:(NSString *)token
+                     sentryDSN:(nullable NSString *)sentryDSN
++         stripePublishableKey:(NSString *)stripePublishableKey
+              googleMapsAPIKey:(nullable NSString *)googleAPIKey
+                    gravityURL:(NSString *)gravity
+                metaphysicsURL:(NSString *)metaphysics
+                     userAgent:(NSString *)userAgent;
+```
+
+`AREmission`'s implementation (.m) needs to be configured to take this new key- It will be exported to our our React Native components as `Emission`. We make our intializer match the signature defined in the header (.h) file, and add an instance `_stripePublishableKey` to match the `@property` declaration.
```

Comment:
our our

Reply:
There's some typos from Danger in here: https://github.com/artsy/artsy.github.io/pull/458/checks